### PR TITLE
Added a command registration service to eliminate the need for a monolith switch binding

### PR DESCRIPTION
**Pull request by Drise13**

### CODE REVIEW

File: BoxwriterResmarkInterop.sln.DotSettings
Comment (Drise13) [inline BoxwriterResmarkInterop.sln.DotSettings:1]: Resharper settings from Control

**Review by agehrin50 — COMMENTED** [review]
Fix possible null warnings

**Review by agehrin50 — APPROVED** [review]

**Comment by Drise13** [comment]
This could also eliminate the need for a list of string constants and the binding provides that as part of the request type

**Comment by Gishe** [comment]
I understand the intent here, but I think it adds more complication and makes the example less digestible.  It does make adding commands a little easier, but I think having the switch statement keeps the decision logic in a single location rather than relying on reflection and registration.

**Pull request closed without merging**